builder/cloudstack/step_create_instance.go

```diff
@@ -149,6 +149,20 @@ func (s *stepCreateInstance) Run(ctx context.Context, state multistep.StateBag)
 	// Store the instance ID so we can remove it later.
 	state.Put("instance_id", instance.Id)
 
+	// Set instance tags
+	if config.Tags != nil {
```

Comment:
When not set, is config.Tags definitely a nil map, or is it just an empty map?  I can imagine this check passing if tags is empty; example: https://play.golang.org/p/EA8lq7Bpdyh

Reply:
One alternative could be to use len() 🙂